docs/modules/ROOT/pages/api/erc1155.adoc

```diff
@@ -0,0 +1,664 @@
+
+[.contract-item]
+[[ERC1155Component-is_approved_for_all]]
+==== `[.contract-item-name]#++is_approved_for_all++#++(self: @ContractState, owner: ContractAddress, operator: ContractAddress) -> bool++` [.item-kind]#external#
+
+Queries if `operator` is an authorized operator for `owner`.
+
+[.contract-item]
+[[ERC1155Component-uri]]
+==== `[.contract-item-name]#++uri++#++(self: @ContractState, token_id: u256) -> ByteArray++` [.item-kind]#external#
+
+This implementation returns the same URI for *all* token types. It relies
+on the token type ID substitution mechanism defined in the EIP:
+https://eips.ethereum.org/EIPS/eip-1155#metadata.
+
+Clients calling this function must replace the `\{id\}` substring with the
+actual token type ID.
+
+==== camelCase Support
+
+[.contract-item]
+[[ERC1155Component-balanceOf]]
+==== `[.contract-item-name]#++balanceOf++#++(self: @ContractState, account: ContractAddress, tokenId: u256) → u256++` [.item-kind]#external#
+
+See <<ERC1155Component-balance_of,ERC1155Component::balance_of>>.
+
+[.contract-item]
+[[ERC1155Component-balanceOfBatch]]
+==== `[.contract-item-name]#++balanceOfBatch++#++(self: @ContractState, accounts: Span<ContractAddress>, tokenIds: Span<u256>) → Span<u256>++` [.item-kind]#external#
+
+See <<ERC1155Component-balance_of_batch,ERC1155Component::balance_of_batch>>.
+
+[.contract-item]
+[[ERC1155Component-safeTransferFrom]]
+==== `[.contract-item-name]#++safeTransferFrom++#++(ref self: ContractState, from: ContractAddress, to: ContractAddress, tokenId: u256,  value: u256, data: Span<felt252>)++` [.item-kind]#external#
+
+See <<ERC1155Component-safe_transfer_from,ERC1155Component::safe_transfer_from>>.
+
+[.contract-item]
+[[ERC1155Component-safeBatchTransferFrom]]
+==== `[.contract-item-name]#++safeBatchTransferFrom++#++(ref self: ContractState, from: ContractAddress, to: ContractAddress, tokenIds: Span<u256>, values: Span<u256>, data: Span<felt252>)++` [.item-kind]#external#
+
+See <<ERC1155Component-safe_batch_transfer_from,ERC1155Component::safe_batch_transfer_from>>.
+
+[.contract-item]
+[[ERC1155Component-isApprovedForAll]]
+==== `[.contract-item-name]#++isApprovedForAll++#++(self: @ContractState, owner: ContractAddress, operator: ContractAddress) -> bool++` [.item-kind]#external#
+
+See <<ERC1155Component-is_approved_for_all,ERC1155Component::is_approved_for_all>>.
+
+[.contract-item]
+[[ERC1155Component-setApprovalForAll]]
+==== `[.contract-item-name]#++setApprovalForAll++#++(ref self: ContractState, operator: ContractAddress, approved: bool)++` [.item-kind]#external#
+
+See <<ERC1155Component-set_approval_for_all,ERC1155Component::set_approval_for_all>>.
+
+==== Internal functions
+
+[.contract-item]
+[[ERC1155Component-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, uri: ByteArray)++` [.item-kind]#internal#
+
+Initializes the contract by setting the token uri as `uri`, and registering the supported interfaces.
```

Comment:
```suggestion
Initializes the contract by setting `uri` as the base URI for all tokens, and registering the supported interfaces.
```
i think we might need to update this variable name to `base_uri` across the codebase, especially the initializer

Reply:
Updated, let me know if you think there are places where we should still update it.

Reply:
Updated!